### PR TITLE
Lazy load dependencies

### PR DESCRIPTION
**Pull request by nikhilkh**

### CODE REVIEW

**Comment by nikhilkh** [comment]
Alternative approach to #81. Fixes #80

**Comment by SBoudrias** [comment]
Really not a fan of lazy loading dependencies... cc @sindresorhus

**Comment by nexdrew** [comment]
@nikhilkh Unfortunately this will need to be rebased.

@SBoudrias I'm usually not a fan of lazy loading deps either, but, due to the nature of this module ("extra" functionality that is non-critical to a CLI, goal of being non-intrusive), I think it's not a bad idea. Hey, at least he chose Sindre's `lazy-req` for the implementation. 😃

File: package.json [hunk 1]
```diff
@@ -40,7 +40,8 @@
     "is-npm": "^1.0.0",
     "latest-version": "^2.0.0",
     "semver-diff": "^2.0.0",
-    "xdg-basedir": "^2.0.0"
+    "xdg-basedir": "^2.0.0",
+    "lazy-req" : "^1.1.0"
```
Comment (nexdrew) [inline package.json]: If we _do_ eventually merge this, we should put this dependency in alphabetical order. Typically npm will do this for us via `npm install --save lazy-req`.

**Comment by nikhilkh** [comment]
I've updated the PR. I wouldn't necessarily lean on lazy loading either but for a module that runs on startup to provide extra features for a CLI - it's important to have minimal impact on startup perf. #80 describes the impact in the current state. Loading `latest-version`, `chalk` etc and their dependencies add undue startup time overhead without any benefit. #80 shows the perf impact on a warm scenario. The impact is even more significant in a cold load scenario,

**Comment by sindresorhus** [comment]
Not a big fan either @SBoudrias, but I've seen `update-notifier` as a require time offender when profiling various things. The idea of update-notifier is that it should have no impact on the consumer. It's just an added bonus, not critical.

**Comment by sindresorhus** [comment]
Changes looks good to me, but should be manually tested before doing a release to ensure it didn't break anything.

**Comment by nexdrew** [comment]
@sindresorhus Worth adding a test that verifies an `update-notifier` dependency does not show up in the `require.cache` until after some logic has run? Or is that just a duplication of `lazy-req` testing?

**Comment by nexdrew** [comment]
@sindresorhus Ok, gotcha. That's what I do when manually testing as well.

**Comment by sindresorhus** [comment]
Opened an issue: https://github.com/yeoman/update-notifier/issues/88

**Comment by SBoudrias** [comment]
@nikhilkh hey, any chance you can fix the merge conflict? Given the other maintainers are fine with going the lazy-req road, I think this would be mergeable as is.

**Comment by nikhilkh** [comment]
Fixed merge conflict. Thanks! It will be great to have this merged soon to avoid future re-basing. :)

**Comment by SBoudrias** [comment]
Thanks!

**Pull request merged**